app/controllers/api/v1/posts_controller.rb

```diff
@@ -4,29 +4,38 @@ module Api
   module V1
     class PostsController < Api::V1::BaseController
       before_action :find_user, only: %i[index create]
+      before_action :find_post, only: %i[show update]
 
       def index
         @posts = @user.posts
       end
 
-      def show
-        @post = Post.find(params[:id])
-      rescue ActiveRecord::RecordNotFound
-        render json: { message: 'Post does not exist' }, status: :not_found
-      end
+      def show; end
 
       def create
-        @post = @user.posts.build(params.require(:post).permit(:title, :body))
+        @post = @user.posts.build(post_params)
         @post.save if validate_post(@user, @post)
       end
 
+      def update
+        return unless validate_not_empty_attriutes(post_params[:title], post_params[:body])
+
+        @post.update(post_params)
+        render 'api/v1/posts/create', formats: [:json], locals: { post: @post }
+      end
+
       # :reek:ControlParameter
       def validate_post(user, post)
         if user != current_user
           render json: { message: 'Unauthorized' }, status: :unauthorized
           return false
-        elsif !post.valid?
-          render json: { errors: post.errors }, status: :bad_request
+        end
+        validate_not_empty_attriutes(post.title, post.body)
+      end
+
+      def validate_not_empty_attriutes(title, body)
+        if !title.present? || !body.present?
```

Comment:
As rule of thumb, we always want to go for the "positive" check (instead of using `!`s)
```suggestion
        unless title.present? && body.present?
```